packages/react-testing/src/root.tsx

```diff
@@ -88,10 +91,20 @@ export class Root<Props> implements Node<Props> {
   ) {
     this.render = render;
     this.resolveRoot = resolveRoot;
+    this.actingPromise = new Promise((resolve) => {
+      this.actingPromiseResolver = resolve;
+    });
     this.mount();
   }
 
   act<T>(action: () => T, {update = true} = {}): T {
+    if (this.destroyed) {
```

Comment:
What purpose is the `destroyed` boolean serving that we don't currently get with the `withRoot` behaviour? It seems that whenever `destroyed` gets set to true, root will also be made unavailable, so we should be able to rely on that existing pattern. In most (all?) cases, act is already prefaced with a `withRoot` check

Reply:
This is mostly for backwards compatibility. I wanted existing code which relied on being able to perform `act` on a destroyed root (see reply above) to function as it did before. 

As for re-using `withRoot` the issue with that is `this.root` being null is set as both the initial value and when it is destroyed and I need to make a distinction between un-initialized vs destroyed thus the new property.